src/AuthManager.ts

```diff
@@ -50,6 +50,40 @@ export class AuthManager {
         return { newCodeVerifier, newCodeChallenge };
     };
 
+    private async refreshAccessToken(): Promise<void> {
+        return new Promise((resolve,reject) => {
+            try {
+                const refreshToken = localStorage.getItem('refresh_token');
+                fetch(`${this.authServer}auth/refresh`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify({
+                        refresh_token: refreshToken,
+                    }),
+                })
+                    .then((response) => {
+                        if (response.status !== 200) {
+                            throw new Error('Failed to refresh the token');
```

Comment:
You can use the `reject()` here. and remove the `try-catch` block

Reply:
Try/catch is important, as it also protect against network errors